src/reduction/mod.rs

```diff
@@ -199,10 +201,15 @@ pub fn reduce_color_type(
 
     // Go down one step at a time
     // Maybe not the most efficient, but it's safe
-    if grayscale_reduction && matches!(reduced.ihdr.color_type, ColorType::RGBA | ColorType::RGB) {
+    if grayscale_reduction
+        && matches!(
+            reduced.ihdr.color_type,
+            ColorType::RGBA | ColorType::RGB { .. }
+        )
+    {
         if let Some(r) = reduce_rgb_to_grayscale(&reduced) {
             reduced = Cow::Owned(r);
-            should_reduce_bit_depth = reduced.ihdr.color_type == ColorType::Grayscale;
+            should_reduce_bit_depth = reduced.ihdr.color_type != ColorType::GrayscaleAlpha;
```

Comment:
I'm not sure why this comparison was changed. What's the rationale behind this change?

Reply:
It was easier than changing it to a `matches!`, which is required to compare with Grayscale 😂
At this point we know it's either Gray or GrayAlpha, so the result is still the same. It's perhaps a bit of an anti-pattern though. Maybe I should add helper functions to `ColorType` to do common matches for rgb/gray/alpha...

Reply:
I've done some additional refactoring here to tidy up and simplify things. The logic is still ultimately the same.

Reply:
In my opinion, It looks much better now. Nice! :smile: